### PR TITLE
Added doc to the progress control

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Complete [taipy#692](https://github.com/Avaiga/taipy/issues/692)

### CODE REVIEW

File: docs/refmans/gui/viselements/generic/progress.md_template [hunk 1]
```diff
@@ -3,26 +3,193 @@ A control that can show the progression of some operation.
...
+The progress indicator represents, in a compact way, the ratio of an achievement on a scale from 0
+to 100 percent.<br/>
+It is useful in situations where a lengthy process is underway, and the application needs to reflect
+how much of the task has been completed at any given moment.
 
-# Styling
+The progress indicator represents, in a compact way, the ratio of achievement on a scale from 0 to
+100 percent.<br/>
+It is useful in situations where a lengthy process is underway, and the application needs to reflect
+how much of the task has been completed at any given moment.
```
Comment (FredLL-Avaiga) [inline docs/refmans/gui/viselements/generic/progress.md_template:9]: again ? cf. line 6-9

File: docs/refmans/gui/viselements/generic/progress.md_template [hunk 2]
```diff
@@ -3,26 +3,193 @@ A control that can show the progression of some operation.
...
+different locations, depending on the setting of the [*title_anchor*](#p-title_anchor) property.
 
 # Usage
 
-## Circular Progress
+## Circular progress indicator {data-source="gui:doc/examples/controls/progress-simple.py"}
+
+The simplest and most compact use case for the progress control is a circular progress
+indicator.<br/>
+How much of the indicator's ring is filled reflects the percentage of achieved progress.
```
Comment (FredLL-Avaiga) [inline docs/refmans/gui/viselements/generic/progress.md_template:30]: should the indeterminate mode be mentioned ?

File: tools/_setup_generation/step_refman.py [hunk 3]
```diff
@@ -382,8 +382,8 @@ def generate_entries(
                             name,
                             TYPE_ID,
                             package,
-                            entry_info["module"],
-                            entry_info.get("packages"),
+                            type["module"],
```
Comment (FredLL-Avaiga) [inline tools/_setup_generation/step_refman.py:385]: why not type.get("module") ?

**Review by FredLL-Avaiga — APPROVED** [review]

File: docs/refmans/gui/viselements/generic/progress.md_template (hunk 2 above)
Comment (FabienLelaquais) [reply docs/refmans/gui/viselements/generic/progress.md_template:30]: Right
Thx

File: tools/_setup_generation/step_refman.py (hunk 3 above)
Comment (FabienLelaquais) [reply tools/_setup_generation/step_refman.py:385]: Any significant difference at this point?

File: docs/refmans/gui/viselements/generic/progress.md_template (hunk 1 above)
Comment (FabienLelaquais) [reply docs/refmans/gui/viselements/generic/progress.md_template:9]: Thx

File: tools/_setup_generation/step_refman.py (hunk 3 above)
Comment (FredLL-Avaiga) [reply tools/_setup_generation/step_refman.py:385]: For consistency sake
Get() would never throw

File: docs/refmans/gui/viselements/generic/progress.md_template (hunk 2 above)
Comment (FabienLelaquais) [reply docs/refmans/gui/viselements/generic/progress.md_template:30]: Done.

File: tools/_setup_generation/step_refman.py (hunk 3 above)
Comment (FabienLelaquais) [reply tools/_setup_generation/step_refman.py:385]: Actually, this is a build-time error. 
We really want the thing to break should this happen...

**Review by FredLL-Avaiga — APPROVED** [review]

**Pull request merged**